syntax = "proto3";
package WorkerData;

service Greeter {
  rpc Call (WorkerRequest) returns (WorkerReply) {}
  rpc SetStatus (WorkerInfo) returns (CommonReply) {}
}

message WorkerRequest {
  string opr = 1;
  string domain = 2;
  string version = 3;
  string resource = 4;
}

message WorkerReply {
  string name = 1;
  string id = 2;
  string host_ip = 3;
  int32 host_port = 4;
  int32 return_code = 5;
}

message WorkerInfo {
  string name = 1;
  string id = 2;
  int32 status = 3;
}

message CommonReply {
  int32 return_code = 1;
}
